examples/sktime/flavor.py

```diff
@@ -0,0 +1,639 @@
+"""The ``flavor`` module provides an example for a custom model flavor for ``sktime`` library.
+
+This module exports ``sktime`` models in the following formats:
+
+sktime (native) format
+    This is the main flavor that can be loaded back into sktime, which relies on pickle
+    internally to serialize a model.
+mlflow.pyfunc
+    Produced for use by generic pyfunc-based deployment tools and batch inference.
+
+    The `pyfunc` flavor of the model supports sktime predict methods `predict`,
+    `predict_interval`, `predict_proba`, `predict_quantiles`, `predict_var`.
+
+    The interface for utilizing a sktime model loaded as a `pyfunc` type for
+    generating forecasts requires passing an exogenous regressor as Pandas
+    DataFrame to the `pyfunc.predict()` method (an empty DataFrame must be
```

Comment:
Can the implementation just specify `None` as a default and if `None` is the arg state upon calling, create an empty DataFrame? Requiring users to create an empty DataFrame to pass in seems a bit odd.

Reply:
Based on the [pyfunc inference API reference](https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html#inference-api) my understanding is that PyFunc models will support only pandas.DataFrame as input - so we need to pass a dataframe right?

Also, according to [model deplyoment API reference](https://mlflow.org/docs/latest/models.html#deploy-mlflow-models) model scoring requires pandas DataFrame csv or json input representation - how would that work with your proposal?

Reply:
Good catch. Yep, we definitely can't pass an empty payload to inference. Related to our discussion below, if we use the DF as a configuration mechanism, this portion of the example could focus more on "here is how you select a specific inference API by interfacing with the pseudo-abstract 'predict' method and the columns hold your arguments as follows: <x, y, z...>"